src/msg/async/AsyncConnection.cc

```diff
@@ -787,6 +787,12 @@ void AsyncConnection::process()
             dispatch_queue->enqueue(message, message->get_priority(), conn_id);
           }
 
+	  // clean up local buffer references
+          data_buf.clear();
+          front.clear();
+          middle.clear();
+          data.clear();
```

Comment:
Note I just copied this section from the cleanup it does in ```case STATE_OPEN_MESSAGE_HEADER:```. I left that present in case we somehow reset back to reading a message without doing another cleanup, but I could try to get rid of the duplicated work if we think we care.

Reply:
This seems fine, but I wonder if we can do better by using swap() to steal the buffers inside decode_message().  That will avoid touching all the refcounts twice (to increment there and then decrement here)

Reply:
I can look into that but it's a larger change touching all implementations rather than a quick bug-fix so let's do it later?